packages/block-editor/src/components/colors/use-colors.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * External dependencies
+ */
+import memoize from 'memize';
+import { startCase } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Children, cloneElement, useMemo } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import InspectorControls from '../inspector-controls';
+import PanelColorSettings from '../panel-color-settings';
+import { useBlockEditContext } from '../block-edit';
+
+const createComponent = memoize( ( attribute, color ) => ( { children } ) =>
+	// Clone children, setting the style attribute from the color configuration,
+	// if not already set explicitly through props.
+	Children.map( children, ( child ) =>
+		cloneElement( child, {
+			style: { [ attribute ]: color, ...child.props.style },
+		} )
+	)
+);
+
+const createSetColor = memoize( ( setAttributes, name ) => ( newColor ) =>
+	setAttributes( { [ name ]: newColor } )
+);
+
+const InspectorControlsColorPanel = ( { title, colorSettings } ) => (
+	<InspectorControls>
+		<PanelColorSettings
+			title={ title }
+			initialOpen={ false }
+			colorSettings={ colorSettings }
+		/>
+	</InspectorControls>
+);
+
+export default function useColors(
+	colorConfigs,
+	deps = [],
+	panelTitle = __( 'Color Settings' )
```

Comment:
I feel deps should always be the last param (to match react) and maybe instead of `paneTitle` we could have a config object (as a way to add other customizations if needed)

Reply:
>I feel deps should always be the last param (to match react)

It's more about which parameters are used more often. Here you would have had people always passing `undefined`, because they wanted the default `panelTitle`.

>we could have a config object (as a way to add other customizations if needed)

Now that we need children inside the panel and optional contrast checking, I will make it an object and make `deps` go last as some of the object's properties might also have dependencies.